test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -26,111 +32,84 @@
-    ("aten::_native_multi_head_self_attention", datetime.date(9999, 1, 1)),
+    "profiler::_call_end_callbacks_on_jit_fut*",
+    "profiler::_record_function_enter",
+    "aten::_cholesky_helper",
+    "aten::_lstsq_helper",
+    "aten::_syevd_helper",
+    "aten::_linalg_solve_out_helper_",
+    "aten::select_backward",
+    "aten::slice_backward",
+    "aten::diagonal_backward",
+    "aten::rowwise_prune",
+    "aten::adaptive_avg_pool3d_backward",
+    "aten::_embedding_bag_dense_backward",
+    "aten::randperm",
+    "aten::_convolution_nogroup",
+    "aten::miopen_convolution_backward",
+    "aten::miopen_convolution_backward_bias",
+    "aten::miopen_convolution_backward_input",
+    "aten::miopen_convolution_backward_weight",
+    "aten::miopen_convolution_transpose_backward",
+    "aten::miopen_convolution_transpose_backward_input",
+    "aten::miopen_convolution_transpose_backward_weight",
+    "aten::miopen_depthwise_convolution_backward",
+    "aten::miopen_depthwise_convolution_backward_input",
+    "aten::miopen_depthwise_convolution_backward_weight",
+    "prepacked::unpack_prepacked_sizes_conv2d",
+    "prepacked::unpack_prepacked_sizes_linear",
+    "aten::native_multi_head_self_attention",
+    "aten::_native_multi_head_self_attention",
 ]
 
-ALLOW_LIST_COMPILED = [
-    (
-        re.compile(item[0]),
-        item[1],
-        re.compile(item[2]) if len(item) > 2 else None,
-    ) for item in ALLOW_LIST if item[1] >= datetime.date.today()
-]
+def compile_temp_allow_list():
+    output = []
+    for item in TEMPORARY_ALLOW_LIST:
+        deadline = item[1]
+        today = datetime.date.today()
+        interval = deadline - today
+        if deadline > today and deadline - today < MAX_ALLOWED_PERIOD:
+            output.append((re.compile(item[0]), deadline, re.compile(item[2]) if len(item) > 2 else None))
+        if interval >= MAX_ALLOWED_PERIOD:
+            print("Operator foo will skip BC and FC schema checks for {} days, but only "
+                  "{} days of skipping the checks are permitted. It's recommended that the skip date be "
+                  "chosen only to permit the PR to be merged. Once the PR is merged for a couple days "
+                  "the operator no longer needs to skip these checks.".format(interval.days, MAX_ALLOWED_PERIOD.days))
+            sys.exit(1)
+
+    return output
+
+TEMPORARY_ALLOW_LIST_COMPILED = compile_temp_allow_list()
 
-def allow_listed(schema):
-    for item in ALLOW_LIST_COMPILED:
+def temp_allow_listed(schema):
+    for item in TEMPORARY_ALLOW_LIST_COMPILED:
         if item[0].search(str(schema)):
             if len(item) > 2 and item[2] is not None:
                 # if arguments regex is present, use it
                 return bool(item[2].search(str(schema)))
             return True
     return False
 
+def indefinite_allow_listed(schema):
+    for item in INDEFINITE_ALLOW_LIST:
+        if re.compile(item).search(str(schema)):
```

Comment:
Maybe we should pre-compile re in `INDEFINITE_ALLOW_LIST` too?

Reply:
Why?